File: Section5 _ViewModel_LiveDataWithDataBinding/CodingChallenge1_java/app/src/main/java/com/comaye/codingchallenge1_java/MainActivityViewModel.java
package com.comaye.codingchallenge1_java;

    import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class MainActivityViewModel extends ViewModel{
//int count = 0;

int count = 0;

public MutableLiveData<Integer> countLiveData = new MutableLiveData<>();

public MutableLiveData<Integer> getCurrentCount() {
    if (countLiveData == null) {
        countLiveData = new MutableLiveData<>();
        countLiveData.setValue(0);
    }

    return countLiveData;
}

public void  increaseCountNumber(){
    count++;
    countLiveData.setValue(count);
}

}

File: Section4_ LiveData/CodingChallenge1/app/src/main/java/com/comaye/codingchallenge1/MainActivityViewModel.kt
package com.comaye.codingchallenge1

import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
class MainActivityViewModel(startingInput : Int): ViewModel(){
//-----------------------------------------------------------------------------------------
   private var count= MutableLiveData<Int>()
    val  countData :LiveData<Int>
    get() = count

    //-----------------------------------------------------------------------------------------
    init {
        count.value = startingInput
    }
    //-----------------------------------------------------------------------------------------
    fun updateCount() {
        count.value = (count.value)?.plus(1)

    }
//-----------------------------------------------------------------------------------------
}